spring-cloud-starter-single-step-batch-job/src/main/java/org/springframework/cloud/task/batch/autoconfigure/rabbit/RabbitAmqpItemWriterAutoConfiguration.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.batch.autoconfigure.rabbit;
+
+import java.util.Map;
+
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
+import org.springframework.batch.item.amqp.AmqpItemWriter;
+import org.springframework.batch.item.amqp.builder.AmqpItemWriterBuilder;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.batch.BatchAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Autconfiguration for a {@code AmqpItemWriter}.
+ *
+ * @author Glenn Renfro
+ * @since 2.3
+ */
+@Configuration
+@EnableConfigurationProperties(RabbitAmqpItemWriterProperties.class)
+@AutoConfigureAfter(BatchAutoConfiguration.class)
+public class RabbitAmqpItemWriterAutoConfiguration {
+
+	@Bean
+	@ConditionalOnProperty(prefix = "spring.batch.job.rabbitamqpitemwriter",
+			name = "name")
+	public AmqpItemWriter<Map<Object, Object>> amqpItemWriter(
+			RabbitTemplate rabbitTemplate) {
+		rabbitTemplate.setMessageConverter(new Jackson2JsonMessageConverter());
```

Comment:
This should be replacable. AKA I should be allowed to bring my own and use the `Jackson2JsonMessageConverter` only if one hasn't been provided.